### ISSUE TITLE
8$: allow '!' after field name with keyword

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#1144]
Example of valid TypeScript code which quick-lint-js rejects:
```typescript
class C {
  public!: boolean;
}
```
